clients/da-vinci-client/src/main/java/com/linkedin/davinci/repository/NativeMetadataRepositoryViewAdapter.java

```diff
@@ -0,0 +1,234 @@
+  }
+
+  @Override
+  public Collection<DerivedSchemaEntry> getDerivedSchemas(String storeName) {
+    return nativeMetadataRepository.getDerivedSchemas(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public DerivedSchemaEntry getLatestDerivedSchema(String storeName, int valueSchemaId) {
+    return nativeMetadataRepository.getLatestDerivedSchema(VeniceView.getStoreName(storeName), valueSchemaId);
+  }
+
+  @Override
+  public RmdSchemaEntry getReplicationMetadataSchema(
+      String storeName,
+      int valueSchemaId,
+      int replicationMetadataVersionId) {
+    return nativeMetadataRepository
+        .getReplicationMetadataSchema(VeniceView.getStoreName(storeName), valueSchemaId, replicationMetadataVersionId);
+  }
+
+  @Override
+  public Collection<RmdSchemaEntry> getReplicationMetadataSchemas(String storeName) {
+    return nativeMetadataRepository.getReplicationMetadataSchemas(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public Store getStore(String storeName) {
+    Store store = nativeMetadataRepository.getStore(VeniceView.getStoreName(storeName));
+    if (store == null) {
+      return null;
+    }
+    if (!VeniceView.isViewStore(storeName)) {
+      return store;
+    }
+    // It's a view store, so we need to create a ReadOnlyViewStore
+    return new ReadOnlyViewStore(store, storeName);
+  }
+
+  @Override
+  public Store getStoreOrThrow(String storeName) throws VeniceNoStoreException {
+    Store store = getStore(storeName);
+    if (store != null) {
+      return store;
+    }
+    throw new VeniceNoStoreException(storeName);
+  }
+
+  @Override
+  public boolean hasStore(String storeName) {
+    return nativeMetadataRepository.hasStore(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public Store refreshOneStore(String storeName) {
+    return nativeMetadataRepository.refreshOneStore(VeniceView.getStoreName(storeName));
+  }
+
+  @Override
+  public List<Store> getAllStores() {
+    return nativeMetadataRepository.getAllStores();
+  }
+
+  @Override
+  public long getTotalStoreReadQuota() {
+    return nativeMetadataRepository.getTotalStoreReadQuota();
+  }
+
+  @Override
+  public void registerStoreDataChangedListener(StoreDataChangedListener listener) {
+    StoreDataChangedListener adapterListener = new StoreDataChangedListenerViewAdapter(listener, this);
+    nativeMetadataRepository.registerStoreDataChangedListener(adapterListener);
+    storeDataChangedAdapterMap.put(listener, adapterListener);
+  }
+
+  @Override
+  public void unregisterStoreDataChangedListener(StoreDataChangedListener listener) {
+    storeDataChangedAdapterMap.compute(listener, (dataChangeListener, dataChangedListenerAdapter) -> {
+      if (dataChangedListenerAdapter != null) {
+        nativeMetadataRepository.unregisterStoreDataChangedListener(dataChangedListenerAdapter);
+      }
+      return dataChangedListenerAdapter;
```

Comment:
Shall we return `null` here so that map will remove this entry?
I have never used `compute` method to remove the entry, so please run some quick verification.